### DISCUSSION TITLE
[Question][Environment Setup]

### DISCUSSION THREAD

**Discussion by seeifsalem** [prosyslab-classroom/cs348-information-security#651] [Q&A]
Name: Mohamed Seifeddine Salem

Hello, I am struggling with the environment setup. I have gone with the docker setup + vs code and followed the instructions but I have a problem. When I reach the step of opening the remote container on vscode it keeps loading saying "opening remote" but nothing happens. I never reach the step where the remote container is indeed open. Do you have any idea of what I should look into to solve this problem?
Thank you for your help

**Comment by m-spitfire** [comment]
#204 is also about same problem, specifically see https://github.com/prosyslab-classroom/cs348-information-security/issues/204#issuecomment-1453252222

**Comment by tanapthetimid** [comment]
I could not get remote container to work either. My work-around was to publish port on docker container and run OpenSSH server inside the docker container then use the SSH remote extension instead.

**Comment by jinho-choi123** [comment]
I used docker-compose instead of just using docker commands.
Here is the instruction I used to setup the project.
https://gitlab.com/-/snippets/2506127

You can just put docker-compose.yml to each project root directory and make separate container!

FYI
I do not recommend using ARM architecture(ex. Apple silicon). It's gonna be really really slow...

**Comment by KihongHeo** [comment]
@seeifsalem your problem solved?
